File: nos2jdbc-tutorial-spring/src/main/java/nos2jdbc/tutorial/spring/service/LunchFeeService.java
package nos2jdbc.tutorial.spring.service;

import java.util.List;

import org.seasar.extension.jdbc.JdbcManager;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import nos2jdbc.tutorial.spring.entity.nonauto.MemberKey;
import nos2jdbc.tutorial.spring.entity.nonauto.Ym;
import nos2jdbc.tutorial.spring.entity.nonauto.rollup.Key;
import nos2jdbc.tutorial.spring.gen.service.LunchFeeServiceBase;

@Service
@Transactional
public class LunchFeeService extends LunchFeeServiceBase {
    @Autowired
    JdbcManager jdbcManager;
    
    public List<Ym> getYm() {
        sqlFilePathPrefix = "sql/";
        return selectBySqlFile(Ym.class, "sum.sql").getResultList();
    }
    public List<MemberKey> getMember() {
        sqlFilePathPrefix = "sql/";
        return selectBySqlFile(MemberKey.class, "sum.sql").getResultList();
    }

    public List<Key> getRollup() {
        sqlFilePathPrefix = "sql/";
        return selectBySqlFile(Key.class, "rollup.sql").getResultList();
    }
    public List<Key> getRollup2() {
        sqlFilePathPrefix = "sql/";
        return selectBySqlFile(Key.class, "rollup2.sql").getResultList();
    }
    public List<Key> getRollup3() {
        sqlFilePathPrefix = "sql/";
        return selectBySqlFile(Key.class, "rollup3.sql").getResultList();
    }
    public List<Key> getCube() {
        sqlFilePathPrefix = "sql/";
        return selectBySqlFile(Key.class, "cube.sql").getResultList();
    }
    public List<Key> getCubeWithJoin() {
        sqlFilePathPrefix = "sql/";
        return selectBySqlFile(Key.class, "cube-with-join.sql").getResultList();
    }
}

File: nos2jdbc-tutorial-kotlin-spring/src/main/kotlin/nos2jdbc/tutorial/kotlinspring/service/LunchFeeService.kt
package nos2jdbc.tutorial.kotlinspring.service

import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional
import nos2jdbc.tutorial.kotlinspring.gen.service.LunchFeeServiceBase
import nos2jdbc.tutorial.kotlinspring.entity.nonauto.Ym
import nos2jdbc.tutorial.kotlinspring.entity.nonauto.MemberKey
import nos2jdbc.tutorial.kotlinspring.entity.nonauto.rollup.Key

@Service
@Transactional
class LunchFeeService: LunchFeeServiceBase() {
    fun getYm() : List<Ym> {
        sqlFilePathPrefix = "sql/";
        return selectBySqlFile(Ym::class.java, "sum.sql").getResultList();
    }
    fun getMember() : List<MemberKey> {
        sqlFilePathPrefix = "sql/";
        return selectBySqlFile(MemberKey::class.java, "sum.sql").getResultList();
    }

    fun getRollup() : List<Key> {
        sqlFilePathPrefix = "sql/";
        return selectBySqlFile(Key::class.java, "rollup.sql").getResultList();
    }
    fun getRollup2() : List<Key> {
        sqlFilePathPrefix = "sql/";
        return selectBySqlFile(Key::class.java, "rollup2.sql").getResultList();
    }
    fun getRollup3() : List<Key> {
        sqlFilePathPrefix = "sql/";
        return selectBySqlFile(Key::class.java, "rollup3.sql").getResultList();
    }
    fun getCube() : List<Key> {
        sqlFilePathPrefix = "sql/";
        return selectBySqlFile(Key::class.java, "cube.sql").getResultList();
    }
    fun getCubeWithJoin() : List<Key> {
        sqlFilePathPrefix = "sql/";
        return selectBySqlFile(Key::class.java, "cube_with_join.sql").getResultList();
    }
}